apps/mobile-wallet/src/persistent-storage/wallet.ts

```diff
@@ -283,23 +213,42 @@ export const storeIsNewWallet = async (isNew: boolean) => {
   }
 }
 
-export const initializeWalletFromStorage = async () => {
-  const decryptedMnemonic = await SecureStore.getItemAsync(MNEMONIC_V2, defaultSecureStoreConfig)
+export const migrateDeprecatedMnemonic = async (deprecatedMnemonic: string) => {
+  // Step 1: Store mnemonic as Uint8Array in secure store without authentication required (as per Uniswap)
+  const mnemonicUint8Array = keyring.importMnemonicString(deprecatedMnemonic)
 
-  if (!decryptedMnemonic) throw new Error('Could not initialize wallet: could not find stored wallet')
+  try {
+    await storeWalletMnemonic(mnemonicUint8Array)
 
-  const parsedDecryptedMnemonic = mnemonicJsonStringifiedObjectToUint8Array(JSON.parse(decryptedMnemonic))
-  keyring.initFromDecryptedMnemonic(parsedDecryptedMnemonic, '')
-}
+    // Step 2: Delete old mnemonic records
+    await SecureStore.deleteItemAsync(PIN_WALLET_STORAGE_KEY, defaultSecureStoreConfig)
```

Comment:
If an error is thrown when deleting records, it will get forwarded (see what's inside the `catch`) and eventually displayed to the user. Same if an error is thrown when storing a new record. Besides, the store methods in this function have their own try/catch which reports the error in the analytics before forwarding it.

Reply:
Alright. Thanks for your answer!